enhancing/enhancer.js

```diff
@@ -6,17 +7,77 @@ module.exports = {
 };
 
 function succeed(item) {
-  return { ...item };
+  if (typeof(item) !== 'object') {
+    return 'error: input must be an object';
+  }
+  let {name,durability,enhancement} = item;
+  if (!name || typeof(durability) !== 'number' || typeof(enhancement) !== 'number') {
+    return 'error: object must include name, durability, and enhancement properties';
+  }
+  
+  if (enhancement < 20) enhancement +=1;
+
+  const success = {
+    name:name,
+    durability:durability,
+    enhancement: enhancement
+  };
+  return { ...success };
 }
 
 function fail(item) {
-  return { ...item };
+  if (typeof(item) !== 'object') {
+    return 'error: input must be an object';
+  }
+  let {name,durability,enhancement} = item;
+  if (!name || typeof(durability) !== 'number' || typeof(enhancement) !== 'number') {
+    return 'error: object must include name, durability, and enhancement properties';
+  }
+
+  if (enhancement < 15) durability -=5;
+  if (enhancement >= 15) durability -=10;
+  if (enhancement > 16) enhancement -=1;
+
+  const failure = {
+    name:name,
+    durability:durability,
+    enhancement: enhancement
+  };
+  return { ...failure };
 }
 
 function repair(item) {
-  return { ...item };
+  if (typeof(item) !== 'object') {
+    return 'error: input must be an object';
+  }
+  const {name,enhancement} = item;
+  if (!name || typeof(enhancement) !== 'number') {
+    return 'error: object must include name and enhancement properties';
+  }
+
+  const repaired = {
+    name:name,
+    durability:100,
+    enhancement:enhancement
+  };
+  return { ...repaired };
 }
 
 function get(item) {
-  return { ...item };
+  if (typeof(item) !== 'object') {
+    return 'error: input must be an object';
+  }
+  let {name,durability,enhancement} = item;
+  if (!name || typeof(durability) !== 'number' || typeof(enhancement) !== 'number') {
+    return 'error: object must include name, durability, and enhancement properties';
+  }
+
+  if (enhancement > 0) name = `[+${enhancement}] ${name}`;
```

Comment:
extra _extra_ edge case: what if the passed-in item has already been modified like this? you wouldn't want `[+7] [+7] iron dagger` for example